src/tools/execution/execution_windows.cpp

```diff
@@ -0,0 +1,91 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Provides simple client MsQuic example that leverages custom execution.
+
+--*/
+
+#define QUIC_API_ENABLE_PREVIEW_FEATURES 1
+
+#include "msquic.hpp"
+#include <stdio.h>
+
+void PrintUsage()
+{
+    printf(
+        "\n"
+        "quicexec is a simple app that can connect to an HTTP/3 server.\n"
+        "\n"
+        "Usage:\n"
+        "\n"
+        "  quicexec <host or ip>\n"
+        );
+}
+
+int
+QUIC_MAIN_EXPORT
+main(
+    _In_ int argc,
+    _In_reads_(argc) _Null_terminated_ char* argv[]
+    )
+{
+    MsQuicApi _MsQuic;
+    if (!_MsQuic.IsValid()) { return 1; }
+    MsQuic = &_MsQuic;
+
+    QUIC_STATUS Status;
+    HANDLE IOCP = CreateIoCompletionPort(INVALID_HANDLE_VALUE, nullptr, 0, 1);
+    QUIC_EXECUTION_CONTEXT_CONFIG ExecConfig = { 0, 0, &IOCP };
+    QUIC_EXECUTION_CONTEXT* ExecContext = nullptr;
+    if (QUIC_FAILED(Status = MsQuic->ExecutionCreate(QUIC_EXECUTION_CONFIG_FLAG_NONE, 1, &ExecConfig, &ExecContext))) {
```

Comment:
how may different binaries are expected to use this interface? i'm wondering about how layering will work, if, say, an app is built on top of an http library on top of msquic.dll.

should the app also open a handle directly to the msquic execution library, or should the http library create a thin wrapper around this, e.g. by typedefing a struct equivalent to the QUIC_CQE struct?

Reply:
high-level point: if the intent of this interface is to allow arbitrary code within a process to efficiently share an execution context, IO completions, timers, etc., then how are component/binaries boundaries supposed to interact - with each other, or directly with the lowest-level execution scheduler?

Reply:
> high-level point: if the intent of this interface is to allow arbitrary code within a process to efficiently share an execution context, IO completions, timers, etc., then how are component/binaries boundaries supposed to interact - with each other, or directly with the lowest-level execution scheduler?
> 
> I don't understand the question.

You've already answered the question here, which is whether or not all binaries/layer should directly register with this EC library, or register with the layer directly below them. I generally agree they should register with the layer below, not directly with this library.